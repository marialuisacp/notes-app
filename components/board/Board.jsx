import React from 'react';
import StyledBoard from './styles/Board.styles';
import BoardHeaderMenu from '../boardHeaderMenu/BoardHeaderMenu';
import CardList from '../cardList/CardList';
import PropTypes from 'prop-types';

const Board = ({ data }) => (
  <div>
    <BoardHeaderMenu></BoardHeaderMenu>
    <StyledBoard>
      {data && data.map((item, key) =>
        <CardList key={key} list={item.cards} title={item.title}></CardList>
      )}
    </StyledBoard>
  </div>
);

Board.propTypes = {
  data: PropTypes.arrayOf(PropTypes.object).isRequired
};

export default Board;