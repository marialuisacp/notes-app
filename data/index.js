exports.data = [
  {
    title: 'Backlog',
    cards: [
      { text: 'Rodar um teste de avaliação do Front-end. Por exemplo: Lightroom', labels: [], icons: [] },
      { text: 'Criação da tela inicial do trello', labels: [], icons: [] },
      { text: 'Modelagem do DB da aplicação', labels: [], icons: [] },
      { text: 'Criação das APIs para consumir os cards', labels: [], icons: [] },
      { text: 'Refatorar o código', labels: [], icons: [] },
    ]
  },
  {
    title: 'Tarefas para fazer hoje',
    cards: [
      { text: 'Criar a estrutura do projeto', labels: ['purple'], icons: [] },
      { text: 'Definir padrões estruturais do projeto', labels: [], icons: [] },
      { text: 'Configuração do servidor onde o projeto será hospedado. Exemplo: Firebase', labels: ['green'], icons: [] },
      { text: 'Instalação e configuração inicial', labels: [], icons: [] },
      { text: 'Configuração do projeto utilizndo CircleCI, contendo toda a pipeline de deploy', labels: [], icons: [] },
      { text: 'Estudar o tema: Cleancode: Nomeclatura de variáveis e complexidade ciclomática', labels: ['blue', 'yellow'], icons: [] },
    ]
  },
  {
    title: 'Em andamento',
    cards: [
      { text: 'Leitura da documentação inicial do Next.js', labels: ['blue'], icons: [] },
      { text: 'Instalação e teste inicial do styled-components', labels: [], icons: [] },
      { text: 'Definição dos componentes iniciais do projeto e da estrutura de pastas e diretórios', labels: [], icons: [] },
    ]
  },
  {
    title: 'Pronto',
    cards: [
      { text: 'Criação do repositório do projeto, escrita do README.md e configurações iniciais', labels: [], icons: [] },
      { text: 'Configuração do Babel com o jest (transfomação dos arquivos de teste para versões mais novas do js)', labels: ['green'], icons: [] },
      { text: 'Leitura de documentação sobre Next.js', labels: ['blue'], icons: [] },
    ]
  },
  {
    title: 'Tarefas impedidas',
    cards: [
      { text: 'Configuração do CircleCI', labels: ['green'], icons: [] },
    ]
  },
  {
    title: 'Links importantes',
    cards: [
      { text: 'Documentação Next.js', labels: ['red'], icons: [] },
      { text: 'Documentação Styled-Components', labels: ['red'], icons: [] },
      { text: 'CircleCI', labels: ['red'], icons: [] },
      { text: 'Documentação React Hooks', labels: ['red'], icons: [] },
      { text: 'Documentação React Testing Library', labels: ['red'], icons: [] },
    ]
  }
];
