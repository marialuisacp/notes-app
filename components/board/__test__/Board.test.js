import React from 'react';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import Board from '../Board';
import { data } from '../../../data';

const defaultProps = {
  data: data
};

test('should render component Board', () => {
  const { BoardFragment } = render(<Board {...defaultProps}></Board>);
  expect(BoardFragment).toMatchSnapshot();
})

test('should have default text in component Board', () => {
  render(<Board {...defaultProps}></Board>);
  expect(screen.getByText(defaultProps.data[0].cards[0].text)).toBeInTheDocument();
})
