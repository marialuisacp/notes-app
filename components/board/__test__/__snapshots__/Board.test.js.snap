// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component Board 1`] = `undefined`;
